templates/config.json.j2

```diff
@@ -1,25 +1,27 @@
 {
-    "masters": [{% for host in groups['mesosmaster'] %}"{{hostvars[host]['ansible_default_ipv4']['address']}}:5050"{% if not loop.last %},{% endif %}{% endfor %}],
-    "refreshSeconds": 60,
-    "ttl": 60,
+    "zk": "zk://{{ mesos_dns_zookeepers | join(',')}}",
+    "zkDetectionTimeout": {{mesos_dns_zkDetectionTimeout}},
+    "masters": {{mesos_masters | to_json}},
+    "refreshSeconds": {{mesos_dns_refreshSeconds}},
+    "ttl": {{mesos_dns_ttl}},
     "domain": {{mesos_dns_domain | to_json}},
     "port": {{mesos_dns_dnsport | to_json}},
-{% if mesos_dns_upstream_resolvers is defined %}
+{% if mesos_dns_upstream_resolvers is defined and mesos_dns_upstream_resolvers | length > 0 %}
```

Comment:
checking first if upstream resolvers are defined else falling back to anbsible nameservers